### DISCUSSION TITLE
American English vs British English

### DISCUSSION THREAD

**Discussion by seancolsen** [mathesar-foundation/mathesar#891] [Developer Q&A] [answered]
In code, UI, and documentation, should we use American English or British English?

Should I name my component `LabeledCheckbox` or `LabelledCheckbox`? My function `analyze_schema` or `analyse_schema`? My UI heading _"Add Visualization"_ or _"Add Visualisation"_?

As an interesting comparison, some of my experience comes from the CiviCRM project which has the following [standard](https://docs.civicrm.org/dev/en/latest/documentation/style-guide/#spelling-and-punctuation):

> Both U.K. and U.S. English spellings are acceptable; we actually welcome inconsistency around this. CiviCRM is an international project, so its mixing it up is a benefit.

**Comment by seancolsen** [comment]
Personally I dislike every option here 😜. Using a mix makes it harder to configure spell-checking. Using one or the other makes it harder on some people.

So realistically, I'm fine with any option, but I'm wondering if other people have preferences.

**Comment by ghislaineguerin** [comment]
We'd probably have to choose one and make it the standard. British English will apply to most people/ be more international considering that it is used in Australia, Ireland, Singapore, India, etc. However, CCI is an American org so not sure if that has more weight in considering one over the other.

**Comment by dmos62** [comment]
I like the idea of mixing. I like language too much to constrain its use, haha. If I had to choose one or the other, I'd say American English is the standard internet language.

**Comment by kgodey** [comment, marked as answer]
I suggest standardizing on American English for the following reasons:
- I think consistency has slight benefits. I like the spirit of the CiviCRM policy, but I think it's going to be confusing if the same word is spelled differently in different places, especially in the code or UI (I don't think it matters that much in docs).
- Programmers are used to using American English (e.g. CSS uses "color", not "colour") so it makes sense for it to be the standard for code.

**Comment by pavish** [comment]
I agree with @kgodey and I think we should go with American English.

Once we have the i18n implementation, we can support British English additionally for text on the UI if we find value in providing it. But in code, I think we can prefer American English.
